### DISCUSSION TITLE
editing cable termination point

### DISCUSSION THREAD

**Discussion by ewiesmann** [netbox-community/netbox#13146] [Help Wanted!] [answered]
### NetBox version
v3.5.4

### Feature type
Editing cable 

### Use case
Want to unplug a cable from Front Port of device A and plug it into Interface of device B

### Problem
If active termination of a cable is a Front Port I can only assign another Front Port but no Interface

### Help
Appreciated!!!

**Comment by candlerb** [comment]
Cables can have multiple terminations at each end, so in theory you should be able to connect to both a frontport and an interface, even simultaneously. But I think I see your problem. Here is the Cable Edit screen (`/dcim/cables/42/edit/`):

![image](https://github.com/netbox-community/netbox/assets/44789/59852c02-9714-4211-8a67-e88a474633de)

Once you've connected a cable end to either an Interface or a Frontport, that's the only thing you can (re)connect to; if you select a Device which has only another type of port, then no ports are shown in the drop-down.

"Interfaces", "Front ports", "Console ports", "Console server ports" etc are all physical things you can plug a cable into (actually not quite, because Interfaces can be virtual too), but Netbox distinguishes them with separate models, and the UI insists you select the type of thing you're connecting to, before selecting the instance.

For now, I think you're going to have to delete the entire cable, and then recreate it.

Some related discussion about cables with multiple terminations: #10841 (but this is only about when the terminations are all of the same type)

**Comment by ewiesmann** [comment]
Hello candler,
thank you for your quick response.

Unfortunately, the answer doesn't really help us.

I think it makes the most sense to trigger a feature request, or does it seem hopeless because the netbox data model makes this option impossible from the outset?

**Reply by candlerb** [reply]
> Unfortunately, the answer doesn't really help us.

I suggested "delete the entire cable, and then recreate it" - can you use that as a workaround?

The data model doesn't make it impossible, and by all means raise a FR.  However, it would be a big change to the cable editing UI.

I note that the data model allows components of different types with the same name, on the same device (e.g. frontport "1" and interface "1"). That would make an unambiguous display of terminations more difficult.

**Comment by ewiesmann** [comment]
> I suggested "delete the entire cable, and then recreate it" - can you use that as a workaround?

Unfortunately, the workaround you described is (at least for us) a bad option.

Because we are engaged in research, network connections are subject to relatively frequent changes.

So it would be a great help for us to be able to change the "Termination" type as easily as possible...

I think I'll just start a FR. (Is there anything I should take note of?)

**Comment by PaulR282** [comment, marked as answer]
I'm making a list of FR for enhancing the object selection. This is also a good idea to be able to connect to different types.

 #12695

**Comment by candlerb** [comment]
> Unfortunately, the workaround you described is (at least for us) a bad option.

I understand. However until [Netbox 3.3](https://github.com/netbox-community/netbox/releases/v3.3.0), it was the only way to change a cable endpoint!

If you have a lot of them to change, it may make sense to download them with a custom CSV export template (#11762, #12018), delete them, modify the CSV, and re-upload them. I've had to do that before.
